### PR TITLE
Write script_id when creating new ChannelTokens

### PR DESCRIPTION
**Pull request by maureensturgeon**
Updates `find_or_create_channel_token` to take a `script_id` which is used only when creating a new channel_token record. All methods calling `find_or_create_channel_token` are updated accordingly. Note that querying for channel_tokens remains unchanged for now until we have backfilled all the records with empty script_ids.

## Links

- jira ticket: [LP-1842](https://codedotorg.atlassian.net/browse/LP-1842)

## Testing story
Added unit tests & tested locally that starting work on a channel-backed level stores the script id and that channel-backed levels that had work done on them previously load as expected.

## Follow-up work
Next up is writing script_id backfills for the existing channel_token records. For the full list of tasks see https://codedotorg.atlassian.net/browse/LP-1395.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/helpers/levels_helper.rb [hunk 1]
```diff
@@ -178,7 +179,7 @@ def app_options
 
     if (@level.channel_backed? && params[:action] != 'edit_blocks') || @level.is_a?(Javalab)
       view_options(
-        channel: get_channel_for(@level, @user),
+        channel: get_channel_for(@level, @script&.id, @user),
```
Comment (davidsbailey) [inline dashboard/app/helpers/levels_helper.rb:182]: from the Jira epic, it looks like you are planning to add a not-null constraint on script_id in the future. This suggests to me that you'll need to come up with a plan for what to do on lines like this for when `@script` is nil. cases which come to mind include (1) standalone projects, and (2) when viewing a level by id in levelbuilder mode. I'm not sure we have tests covering the latter. I'll see if I can add one.

**Comment by jamescodeorg** [comment]
Do we currently have any levels that are shared between scripts (i.e. do we currently have a "live" instance of this bug)?

**Review by davidsbailey — APPROVED** [review]
Looks good to me!

File: dashboard/app/helpers/levels_helper.rb (hunk 1 above)
Comment (davidsbailey) [reply dashboard/app/helpers/levels_helper.rb:182]: https://github.com/code-dot-org/code-dot-org/pull/42167

**Comment by maureensturgeon** [comment]
> Do we currently have any levels that are shared between scripts (i.e. do we currently have a "live" instance of this bug)?

Yes, check out the comments in this ticket with an example https://codedotorg.atlassian.net/browse/LP-1752

File: dashboard/app/helpers/levels_helper.rb (hunk 1 above)
Comment (maureensturgeon) [reply dashboard/app/helpers/levels_helper.rb:182]: So the ticket with the non-null constraint was resolved with status "won't do" https://codedotorg.atlassian.net/browse/LP-1846, for reason number 2 that you mentioned which is that in levelbuilder we'll be able to view a level that's channel-backed outside of the context of a script so script_id will be nil.

Comment (davidsbailey) [reply dashboard/app/helpers/levels_helper.rb:182]: Great! No concerns with instances of `@script&.id`, then 👍

File: dashboard/test/models/channel_token_test.rb [hunk 2]
```diff
@@ -6,19 +6,49 @@ class ChannelTokenTest < ActiveSupport::TestCase
...
+      level,
+      @fake_ip,
+      storage_id,
+      @script.id
+    )
+
+    assert_equal expected_channel_token.id, channel_token.id
+  end
+
+  test 'find_or_create_channel_token will create a new channel token with script_id if no channel token exists' do
```
Comment (jamescodeorg) [inline dashboard/test/models/channel_token_test.rb:43]: Note for future PR: If we decide that passing in nil as the script_id is a valid scenario, let's add a test for it.

**Review by jamescodeorg — APPROVED** [review]

**Pull request merged**